dashboard/src/actions/repos.ts

```diff
@@ -101,6 +101,30 @@ export const resyncRepo = (
   };
 };
 
+export const resyncAllRepos = (): ThunkAction<Promise<void>, IStoreState, null, AppReposAction> => {
+  return async (dispatch, getState) => {
+    try {
+      const {
+        config: { namespace },
+      } = getState();
+      // TODO: Do something to show progress
+      const repos = await AppRepository.list(namespace);
```

Comment:
I'm not sure we need to refresh the list of repos here, could we not just pass the list of repos we've already fetched from the component, or just get them from the state??

Reply:
In case the backend repo data changed, it's meaningful to get the repos from backend again. Otherwise, for example if the backend repo has repo.spec.resyncRequests increased by 1, then the sync repo action here doesn't take effect.

However if we don't consider this case, it's OK to not to call `AppRepository.list(namespace);`. BTW could you guide me how to get it from the state please? I'm very new to ReactJS :)

Reply:
so, in the component `AppRepoList`, there is a property `repos`. That's already the list of repositories. You can forward that list to `AppRepoRefreshAllButton` and then make this method accept a string array so you don't need to list again.

Reply:
Nice. Let me try that.

Reply:
Done.